src/core/kconfigelektra.cpp

```diff
@@ -0,0 +1,225 @@
+ */
+inline void traverseIterators(Key::iterator* parent, Key::iterator* child) {
+    /*
+     * First item of iterator seems to be namespace, as namespaces can differ with cascading keys, we skip these.
+     * (Loop initializer)
+     *
+     * Loop as long as the paths are the same and iterate through the path segments.
+     */
+    for ((*parent)++, (*child)++; parent->get() == child->get(); (*parent)++, (*child)++);
+}
+
+struct KConfigKey {
+    std::string group;
+    std::string key;
+};
+
+inline KConfigKey elektraKeyToKConfigKey(Key::iterator* key, Key::iterator* end) {
+    std::ostringstream group;
+    std::string key_name;
+    bool first = true;
+
+    for(; (*key) != (*end); (*key)++) {
+        if (!key_name.empty()) {
+            if (!first) {
+                group << "\x1d";
+            } else {
+                first = false;
+            }
+            group << key_name;
+        }
+        key_name = key->get();
+    }
+
+    return KConfigKey {
+        group.str(),
+        key_name,
+    };
+}
+
+KConfigBackend::ParseInfo
+KConfigElektra::parseConfig(const QByteArray& /*locale*/, KEntryMap &entryMap, KConfigBackend::ParseOptions options) {
+    //TODO error handling
+    //TODO properly handle parse options (or figure out how they actually work)
+
+    KDB kdb = KConfigElektra::open_kdb();
+    KeySet keySet;
+
+    bool oGlobal = options & ParseGlobal;
+
+    Key parent_key = Key(app_key());
+
+    kdb.get(keySet, parent_key);
+    keySet = keySet.cut(parent_key);
+
+    Key key;
+
+
+    while((key = keySet.pop()) != nullptr) {
+
+        if (key.isDirectBelow(parent_key)) {
+
+            KEntryMap::EntryOptions entryOptions = nullptr;
+
+            if (oGlobal) {
+                entryOptions |= KEntryMap::EntryGlobal;
+            }
+
+            entryMap.setEntry(
+                    QByteArray::fromStdString("<default>"),
+                    QByteArray::fromStdString(key.getBaseName()),
+                    QByteArray::fromStdString(key.getString()),
+                    entryOptions
+            );
+        } else {
+
+            KEntryMap::EntryOptions entryOptions = nullptr;
+
+            if (oGlobal) {
+                entryOptions |= KEntryMap::EntryGlobal;
+            }
+
+            auto parentIter = parent_key->begin();
+            auto childIter = key.begin();
+            traverseIterators(&parentIter, &childIter);
+
+            auto childEnd = key.end();
+
+            auto configKey = elektraKeyToKConfigKey(&childIter, &childEnd);
+
+            entryMap.setEntry(
+                    QByteArray::fromStdString(configKey.group),
+                    QByteArray::fromStdString(configKey.key),
+                    QByteArray::fromStdString(key.getString()),
+                    entryOptions
+            );
+        }
+    }
+
+    kdb.close();
+
+    return ParseOk;
+}
+
+inline std::string kConfigGroupToElektraKey(std::string group, const std::string& keyname) {
```

Comment:
This might introduce some trouble. In `kconfig` config files key names may contain the `/` character. I used `ckdb::Key.setBaseName` to append key/group names to a `ckdb::Key` so that it escapes characters automatically.

Reply:
okay, changed it

I pushed the changes to the branch of #14

Reply:
maybe lets merge #14 and close this PR. It is a bit confusing now.